### DISCUSSION TITLE
Trouble removing releases from folders

### DISCUSSION THREAD

**Discussion by captain-cronch** [joalla/discogs_client#160] [General]
I'm trying to remove specific releases from specific folders, but have found that the release is being removed from my collection entirely. Is it within the capability of remove_release to specifically remove the release from a given folder and not from the collection as a whole?

**Comment by JOJ0** [comment]
https://www.discogs.com/developers#page:user-collection,header:user-collection-delete-instance-from-folder

is the API python3-discogs-client uses, and that's the docs for the implementation: 

https://python3-discogs-client.readthedocs.io/en/docs_theme/discogs_client.models.html#discogs_client.models.CollectionFolder.remove_release

So you sinple pass the  collection item instance id.

Removing from collection folder id 0 (zero) removes from the whole collection!

Show us some code of yours please!

**Comment by captain-cronch** [comment]
I was essentially using the following thinking that it would remove the release from the specified folder:
```
folder = user.collection_folders[1]
release = folder.releases[0]
folder.remove_release(release)
```
After reading the API, I believe that the function I'm actually hoping to use is:
[https://www.discogs.com/developers#page:user-collection,header:user-collection-change-rating-of-release](url) 
which allows the passing of a second folder_id in the request body (1 for my uncategorized folder, etc). In addition to changing the rating, the documentation states it can also be used to move folders, which is my intended goal (to move a release back to uncategorized, not just remove it from the folder).

Is there anything in the current python3-discogs-client that accomplishes this?

Edit:
I'm successfully able to move releases between folders using requests and the previously mentioned method from the discogs API, but still hoping there's a method in python3-discogs-client so I can simplify my code

**Comment by JOJ0** [comment]
try that


```
instance_id = folder.releases[0].id  # gives you collection item instance id
folder.remove_release(instance_id)

```

**Reply by JOJ0** [reply]
but I just read that you must suplly a whole instance, instance id is not supported, I'm sorry for the mistake: https://python3-discogs-client.readthedocs.io/en/latest/fetching_data.html#removing-a-release-from-a-collection-folder

have you already tried the 3rd of the described ways?

**Reply by captain-cronch** [reply]
The 3rd described way appears to also delete every instance in every folder. I used the following using requests to accomplish what I was after (having already established username from authentication and other necessary variables from user input):

```
url = f"https://api.discogs.com/users/{username}/collection/folders/{folder_id}/releases/{release_id}/instances/{instance_id}"

data = {'folder_id': target_folder_id}

response = requests.post(url, auth=auth, json=data)
```

I looked through models.py and believe there's not currently a function that accomplishes this (perhaps I'm not passing optional arguments somewhere though). In the class CollectionFolder, a function called move_release would be useful and probably look something like this:

```
def move_release(self, instance, target_folder_id):
    if not isinstance(instance, CollectionItemInstance):
        raise TypeError('instance must be of type CollectionItemInstance')
    instance_url = self.fetch('resource_url') + '/releases/{0}/instances/{1}'.format(instance.id, instance.instance_id)
    data = {'folder_id': target_folder_id}
    self.client._post(instance_url, data)
```

As mentioned earlier, I wanted to remove a release from a folder, but to accomplish that it must be moved to a different folder (such as the uncategorized folder). This function also allows the passing of a different target_folder_id and would result in an existing instance being moved from its current folder to the target folder.
